libcaf_core/caf/flow/byte.test.cpp

```diff
@@ -0,0 +1,220 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#include "caf/flow/byte.hpp"
+
+#include "caf/test/caf_test_main.hpp"
+#include "caf/test/fixture/deterministic.hpp"
+#include "caf/test/fixture/flow.hpp"
+#include "caf/test/scenario.hpp"
+#include "caf/test/test.hpp"
+
+#include "caf/flow/multicaster.hpp"
+
+using namespace caf;
+
+namespace {
+
+/// Compares two chunks for equality.
+[[maybe_unused]] bool operator==(const caf::chunk& lhs, const caf::chunk& rhs) {
+  return std::equal(lhs.bytes().begin(), lhs.bytes().end(),
+                    rhs.bytes().begin());
```

Comment:
Unsafe, should use the four-argument overload. However, please don't patch `chunk` at file scope. That makes it hard to understand what's going on. Better: give `chunk` an `equal_to` member function and use that for testing.